src/Dev/Deprecation.php

```diff
@@ -228,24 +250,22 @@ public static function outputNotices(): void
         if (!self::isEnabled()) {
             return;
         }
-        $outputMessages = [];
-        // using a while loop with array_shift() to ensure that self::$userErrorMessageBuffer will have
-        // have values removed from it before calling user_error()
-        while (count(self::$userErrorMessageBuffer)) {
+
+        $count = 0;
+        $origCount = count(self::$userErrorMessageBuffer);
+        while ($origCount > $count) {
+            $count++;
             $arr = array_shift(self::$userErrorMessageBuffer);
             $message = $arr['message'];
-            // often the same deprecation message appears dozens of times, which isn't helpful
-            // only need to show a single instance of each message
-            if (in_array($message, $outputMessages)) {
-                continue;
-            }
```

Comment:
deduplication is moved into `notice()` - doesn't make sense to add messages to the buffer if they're already in there.